### PR TITLE
jenkins: use golangci

### PR DESCRIPTION
**Pull request by greut**
gometalinter being too slow and playing badly with go1.11.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -50,15 +50,13 @@ def gofmt() {
...
 def golint(repo) {
   docker.withRegistry('https://registry.internal.exoscale.ch') {
-    def image = docker.image('registry.internal.exoscale.ch/exoscale/golang:1.10')
+    def image = docker.image('registry.internal.exoscale.ch/exoscale/golang:1.11')
     image.pull()
     image.inside("-u root --net=host -v ${env.WORKSPACE}/src:/go/src/github.com/${repo}") {
-      sh "golint -set_exit_status -min_confidence 0.3  `go list github.com/${repo}/... | grep -v /vendor/`"
       sh "go vet `go list github.com/${repo}/... | grep -v /vendor/`"
-      sh "cd /go/src/github.com/${repo} && gometalinter ./..."
+      sh "cd /go/src/github.com/${repo} && golangci-lint run ./..."
```
[the remark this answers is not in the document]
Comment (greut) [reply Jenkinsfile]: Are you a valid GOPATH? cf. the left hand part of the `sh` invocation.

File: cmd/privnet.go [hunk 2]
```diff
@@ -34,6 +34,26 @@ func getNetworkByName(name string) (*egoscale.Network, error) {
 	return net, nil
 }
 
+func getNetworkByNameAndZone(name string, zoneID *egoscale.UUID) (*egoscale.Network, error) {
```
[the remark this answers is not in the document]
Comment (greut) [reply cmd/privnet.go]: A side effect of fixing the errors reported by _golangci-lint_

```
+ golangci-lint run ./...

cmd/vm_create.go:211:24: `getSecurityGroups` - `cs` is unused (unparam)

func getSecurityGroups(cs *egoscale.Client, commaParameter string) ([]egoscale.UUID, error) {

                       ^

cmd/vm_create.go:228:37: `getPrivnetList` - `zoneID` is unused (unparam)

func getPrivnetList(commaParameter, zoneID string) ([]egoscale.UUID, error) {

                                    ^

cmd/vm_create.go:245:23: `getAffinityGroup` - `cs` is unused (unparam)

func getAffinityGroup(cs *egoscale.Client, commaParameter string) ([]egoscale.UUID, error) {

                      ^

doc/main.go:200:58: `printOptions` - `name` is unused (unparam)

func printOptions(buf *bytes.Buffer, cmd *cobra.Command, name string) error {

                                                         ^

cmd/template.go:128:2: Consider preallocating `keys` (prealloc)

	var keys []string

	^

script returned exit code 1
```

File: Jenkinsfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (chrisglass) [reply Jenkinsfile]: I thought so, but apparently not. Or maybe my other machine is setup in a weird way.

The following steps work:

- go get -u github.com/golangci/golangci-lint/cmd/golangci-lint
- go get -u github.com/exoscale/cli
- cd $GOPATH/src/github.com/golangci/golangci-lint
- go install ./...
- cd $GOPATH/src/github.com/exoscale/cli
- $GOPATH/bin/golangci-lint run ./...

File: cmd/privnet.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (chrisglass) [reply cmd/privnet.go]: Ah :)

**Comment by greut** [comment]
@chrisglass cool, thanks!!

**Pull request merged**